Tasks/Gradle/CodeAnalysis/SonarQube/metrics.ts

```diff
@@ -56,6 +57,15 @@ export class SonarQubeMetrics {
     }
 
     /**
+     * Returns the status of the project under analysis (the quality gate status)
+     * @param analysisDetails A JSON object representation of the analysis
+     * @returns String representing the result of the project analysis
+     */
+    public static getQualityGateStatus(analysisDetails:any):string {
+        return analysisDetails.projectStatus.status;
```

Comment:
The expected way this method is used will be 

_return metricsInstance.getAnalysisDetails().then((analysisDetails) => {
&nbsp;&nbsp;&nbsp;&nbsp;return SonarQubeMetrics.getQualityGateStatus(analysisDetails))
}_

Here, the call to _getAnalysisDetails()_ contains the heavyweight logic of calling, extracting, etc.. The static call to getQualityGateStatus is merely to extract the relevant field from within the analysis details, and is made public for cases where the analysis details have already been fetched.